anime_downloader/util.py

```diff
@@ -78,35 +78,69 @@ def format_search_results(search_results):
     return table
 
 
-def search(query, provider, choice=None):
+def search(query, provider, val=None, season_info=None, ratio=50):
+    # Will use animeinfo sync if season_info is provided
+
     # Since this function outputs to stdout this should ideally be in
     # cli. But it is used in watch too. :(
     cls = get_anime_class(provider)
     search_results = cls.search(query)
-    click.echo(format_search_results(search_results), err=True)
 
     if not search_results:
         logger.error('No such Anime found. Please ensure correct spelling.')
-        sys.exit(1)
-
-    if choice:
-        val = choice
-    else:
-        val = click.prompt('Enter the anime no: ', type=int, default=1, err=True)
+        return None, None
 
-    try:
-        url = search_results[val-1].url
-        title = search_results[val-1].title
-    except IndexError:
-        logger.error('Only maximum of {} search results are allowed.'
-                     ' Please input a number less than {}'.format(
-                         len(search_results), len(search_results)+1))
-        sys.exit(1)
+    if season_info:
+        from anime_downloader import animeinfo
+        match = animeinfo.fuzzy_match_metadata([season_info], search_results)
+        logger.debug('Match ratio: {}'.format(match.ratio))
+        # ratios are a range between 0-100 where 100 means 100% match.
+        if match.ratio >= ratio and not val:
+            logger.debug('Selected {}'.format(match.SearchResult.title))
+            return match.SearchResult.url, None
 
+    click.echo(format_search_results(search_results), err=True)
+    # Loop to allow re-propmt if the user chooses incorrectly
+    # Makes it harder to unintentionally exit the anime command if it's automated
+    while True:
+        if val == None:
+            val = click.prompt('Enter the anime no{}:'. format(' (0 to switch provider)'*(season_info != None)),
+                type=int, default=1, err=True)
+        try:
+            url = search_results[val-1].url
+            title = search_results[val-1].title
+        except IndexError:
+            logger.error('Only maximum of {} search results are allowed.'
+                         ' Please input a number less than {}'.format(
+                             len(search_results), len(search_results)+1))
+            val = False
+            continue
+        break
 
     logger.info('Selected {}'.format(title))
 
-    return url
+    return url, val
```

Comment:
Why not return search result?

Reply:
To save the result for future episodes. You wouldn't wanna enter your choice 12 times if there's 12 episodes.